Docs: What’s new & Upgrade guide 10.4

imatwawana opened the pull request:
**PLEASE DO NOT ADD DIRECTLY TO THIS PR; CREATE A BRANCH AND MAKE CHANGES THERE FOR REVIEW PRIOR TO MERGING**

This PR adds the What’s new, which describes new features and changes available in Grafana v10.4, and the v10.4 Upgrade guide.

**Note for reviewers**: Review should be limited to placement of entries. Please do not make any copy edits. These entries are exactly the same as what's in the What's new in Cloud and making changes here creates divergence. If there's anything that looks wrong or is missing, please flag it that so we can decide on a course of action. 

**TO DO**

- [x] Add enablement videos as they are available - videos added but commented out. Will be commented in on a separate PR when videos are out of draft.

imatwawana committed:
Updated index pages and added v10.4 breaking changes, upgrade guide, and what's new pages

imatwawana committed:
Added what's new entries

imatwawana committed:
Merge branch 'main' into docs/whats-new-v10.4

imatwawana committed:
Removed empty headings

imatwawana committed:
Fixed link

imatwawana committed:
Removed duplicate entry and fixed styling

imatwawana committed:
Removed breaking changes page and references to it

imatwawana committed:
Added intro text

JacobsonMT committed:
Docs: 10.4 technical note for alertingUpgradeDryrunOnStart (#83262)

* Docs: 10.4 technical note for alertingUpgradeDryrunOnStart

* Apply suggestions from code review

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Address PR comments

* restarts -> starts

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

imatwawana commented on docs/sources/upgrade-guide/upgrade-v10.4/index.md:
```diff
@@ -0,0 +1,37 @@
+
+If you haven't already upgraded to Grafana Alerting from legacy Alerting, Grafana will initiate a dry-run of the upgrade every time the instance starts. This is in preparation for the removal of legacy Alerting in Grafana v11. The dry-run logs the results of the upgrade attempt and identifies any issues requiring attention before you can successfully execute the upgrade. No changes are made durin ...
+
+You can disable this behaviour using the feature flag `alertingUpgradeDryrunOnStart`:
```
American English
```suggestion
You can disable this behavior using the feature flag `alertingUpgradeDryrunOnStart`:
```

imatwawana committed:
Fixed spelling

imatwawana commented:
@mjseaman - Note for review: just a reminder that we're not doing any copy editing; these entries (apart from any egregious typos) are exactly the same as what's in the What's new in Cloud and we don't want to create divergence. If there's anything that looks wrong, is missing, etc, then please flag that.

imatwawana committed:
Added new entries to What's new

mjseaman committed:
reorder and add intro

mjseaman commented:
Thanks Isabel. 

I've added a longer intro paragraph and reordered some items for emphasis. I made a change before seeing your note that I should cut a separate PR - sorry! 

Some what's new posts use "you" and others use "users" to address the subject. IMO we should standardize on "you" per the [style guide](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#address-users-clearly). I can go through the what's new posts and edit them there - is that the best course of action?

Also, I think we should include:
- the new Pagerduty data source - even though it's not technically part of the Grafana codebase, it's part of Grafana and this is a good place to announce it. https://admin.grafana.com/whats-new/#pagerduty-enterprise-data-source-for-grafana
- Return to previous. This one was marked for 10.4 - I wonder why it didn't make it into this PR.. Maybe it was submitted late? https://admin.grafana.com/content-admin/#/collections/whats-new/entries/2024-02-21-the-return-to-previous-functionality

I've also poked a couple of stragglers here to make sure significant updates are all in: https://raintank-corp.slack.com/archives/C036J5B39/p1709120845814739?thread_ts=1707928966.392599&cid=C036J5B39

imatwawana commented:
> Thanks Isabel.
> 
> I've added a longer intro paragraph and reordered some items for emphasis. I made a change before seeing your note that I should cut a separate PR - sorry!

Making changes to the order and intro directly on the PR is fine. I just don't want folks adding entries to the PR directly.
> 
> Some what's new posts use "you" and others use "users" to address the subject. IMO we should standardize on "you" per the [style guide](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#address-users-clearly). I can go through the what's new posts in the CMS and edit them there - is that the best course of action?

If you want to make copy edits to the posts, then yes, make those updates in the CMS and I'll implement those updates here when we generate the file for the last time tomorrow or Monday. Yes we should be using "you" vs "users", but that stuff should all be reviewed before folks merge their CMS entries. That's the behaviour we want to encourage: getting review (say during docs office hours, for example) before those entries ever make it to this side of things vs expecting cleanup at this stage where we've decided the docs team isn't going to do that anymore to keep this scalable.
> 
> Also, I think we should include:
> 
> * the new Pagerduty data source - even though it's not technically part of the Grafana codebase, it's part of Grafana and this is a good place to announce it. https://admin.grafana.com/whats-new/#pagerduty-enterprise-data-source-for-grafana

Okay, I'll add that in.

> * Return to previous. This one was marked for 10.4 - I wonder why it didn't make it into this PR.. Maybe it was submitted late? https://admin.grafana.com/content-admin/#/collections/whats-new/entries/2024-02-21-the-return-to-previous-functionality

Yes, that's likely what happened. But we'll generate another file from the What's new in Cloud tomorrow and probably again Monday to catch stragglers. I'll be merging this PR EOD Monday. Anything that comes in later than that will get added maybe about a week after that when I catch the last of the stragglers.
> 
> I've also poked a couple of stragglers here to make sure significant updates are all in: https://raintank-corp.slack.com/archives/C036J5B39/p1709120845814739?thread_ts=1707928966.392599&cid=C036J5B39

imatwawana committed:
Added PagerDuty data source entry

imatwawana committed:
Added entries

imatwawana committed:
Added new entries

imatwawana committed:
Fixed formatting

imatwawana committed:
Fixed page weight and links

imatwawana committed:
Merge branch 'main' into docs/whats-new-v10.4

mjseaman commented on docs/sources/whatsnew/whats-new-in-v10-4.md:
```diff
@@ -0,0 +1,273 @@
+
+# What’s new in Grafana v10.4
+
+Welcome to Grafana 10.4! This minor release contains some notable improvements in its own right, as well as early previews of functionality we intend to turn on by default in Grafana v11. Read on to learn about a quicker way to set up alert notifications, an all-new UI for configuring up single sign-on, and improvements to our Canvas, Geomap, and Table panels.
```
```suggestion
Welcome to Grafana 10.4! This minor release contains some notable improvements in its own right, as well as early previews of functionality we intend to turn on by default in Grafana v11. Read on to learn about a quicker way to set up alert notifications, an all-new UI for configuring single sign-on, and improvements to our Canvas, Geomap, and Table panels.
```

imatwawana committed:
Apply suggestion from review

Co-authored-by: Mitch Seaman <mjseaman@users.noreply.github.com>

mjseaman committed:
remove team lbac, move return to previous (#83921)

- Remove Team LBAC for Loki (it is Cloud only)
- Move Return to Previous into the Alerting section, since it only works for Alerting now
- Add a note that data sources are separate from Grafana but included for visibility

imatwawana commented on docs/sources/whatsnew/whats-new-in-v10-4.md:
```diff
@@ -0,0 +1,256 @@
+
+## Data sources
+
+> Note: the following data sources are released separately from Grafana itself. They are included here for extra visibility.
```
```suggestion
{{< admonition type="note" >}}
The following data sources are released separately from Grafana itself. They are included here for extra visibility.
{{< /admonition >}}
```

imatwawana committed:
Replaced manual note with admonition shortcode

mjseaman committed:
move Return to Previous out of Alerting section

imatwawana committed:
Added youtube links

imatwawana committed:
Commented out youtube videos and removed duplicate video embed
